### DISCUSSION TITLE
Deleting historical data points from a OPCUA server using asyncua

### DISCUSSION THREAD

**Discussion by KishanKishore** [FreeOpcUa/opcua-asyncio#479] [Q&A]
The ```asyncua``` library currently has methods to read historical data from an OPCUA server.
1. According to the OPCUA specification the server can have the following capability. More on this [here](https://reference.opcfoundation.org/v104/Core/docs/Part11/6.8.6/)
```
The DeleteAtTimeCapability Variable indicates support for the delete at time capability. A value of True indicates the Server supports the capability to delete a data value at a specified time. The default value is False.
```
2. Does the ```asyncua``` client support the deletion of history from an OPCUA server ? 
3. Is there a way we can fetch the range of historical data present in an OPCUA server using ```asyncua```?

**Comment by AndreasHeine** [comment]
> Does the asyncua client support the deletion of history from an OPCUA server ?

https://reference.opcfoundation.org/v104/Core/docs/Part4/5.10.5/

**5.10.5 HistoryUpdate:** 

there is currently no method in uaclient.py 
https://github.com/FreeOpcUa/opcua-asyncio/blob/master/asyncua/client/ua_client.py

but all lowlevel classes are avalible to make a request:
https://github.com/FreeOpcUa/opcua-asyncio/blob/398a45c199dc436459be809323e7591082d5c9d5/asyncua/ua/uaprotocol_auto.py#L6265

https://github.com/FreeOpcUa/opcua-asyncio/blob/398a45c199dc436459be809323e7591082d5c9d5/asyncua/ua/uaprotocol_auto.py#L6275

if you have a server which supports that feature you could make a PR to add that feature to the uaclient.py / client.py!

is not implemented in asyncua-server!
all currently supported requesttypesof the server:
https://github.com/FreeOpcUa/opcua-asyncio/blob/398a45c199dc436459be809323e7591082d5c9d5/asyncua/server/uaprocessor.py#L150

> Is there a way we can fetch the range of historical data present in an OPCUA server using asyncua?

the high level client (client.py) no!
the low level client (ua.client.py) yes! 
https://github.com/FreeOpcUa/opcua-asyncio/blob/398a45c199dc436459be809323e7591082d5c9d5/asyncua/client/ua_client.py#L619

**Reply by KishanKishore** [reply]
Thanks for the response. I will have a look at these classes/methods.
